common/src/main/java/tachyon/util/io/BufferUtils.java

```diff
@@ -148,14 +149,19 @@ public static ByteBuffer getIncreasingIntBuffer(int len) {
   }
 
   /**
-   * * Writes buffer to the given file path
+   * Writes buffer to the given file path. If this file already exists, deletes it first.
    *
    * @param path file path to write the data
    * @param buffer raw data
-   * @throws java.io.IOException
+   * @throws java.io.IOException if an I/O error occurred or failed to delete existing file
    */
   public static void writeBufferToFile(String path, byte[] buffer) throws IOException {
-    FileOutputStream os = new FileOutputStream(path);
+    File file = new File(path);
+    if (file.exists()) {
+      FileUtils.delete(file);
+    }
+    FileUtils.createFile(file);
+    FileOutputStream os = new FileOutputStream(file);
```

Comment:
The problem is if parent dir of the target file is not constructed yet, the constructor will fail